src/dsp_tools/commands/xmllib/models/resource.py

```diff
@@ -0,0 +1,175 @@
+
+
+@dataclass
+class Resource:
+    res_id: str
+    restype: str
+    label: str
+    values: list[Value] = field(default_factory=list)
+    permissions: str = "res-default"
+    iiif_uri: str | None = None
+    filepath: str | None = None
+    permissions_for_file: str = "prop-default"
+
+    def __post_init__(self) -> None:
+        if not is_string(str(self.label)):
+            msg = (
+                f"The label of a resource should be a string.\n"
+                f"The label '{self.label}' of the resource with the ID {self.res_id} is not a string."
+            )
+            warnings.warn(DspToolsUserWarning(msg))
+
+    def serialise(self) -> etree._Element:
+        res_ele = self._make_resource_ele()
+        if (file_ele := self._make_file_ele()) is not None:
+            res_ele.append(file_ele)
+        res_ele.extend(self._serialise_values())
+        return res_ele
+
+    def _make_resource_ele(self) -> etree._Element:
+        attribs = {"label": self.label, "restype": self.restype, "id": self.res_id}
+        if self.permissions:
+            attribs["permissions"] = self.permissions
+        return etree.Element(f"{DASCH_SCHEMA}resource", attrib=attribs, nsmap=XML_NAMESPACE_MAP)
+
+    def _make_file_ele(self) -> etree._Element | None:
+        if self.iiif_uri and self.filepath:
+            raise InputError(
+                "A resource can only have a iiif-uri or a filepath.\n"
+                f"The resource with the ID '{self.res_id}' has both."
+            )
+        if self.filepath:
+            prop_ = etree.Element(
+                f"{DASCH_SCHEMA}bitstream",
+                permissions=self.permissions_for_file,
+                nsmap=XML_NAMESPACE_MAP,
+            )
+            prop_.text = str(self.filepath)
+            return prop_
+        if self.iiif_uri:
+            if not is_iiif_uri(self.iiif_uri):
+                msg = (
+                    f"Failed validation in iiif-uri tag of resource '{self.res_id}': "
+                    f"The URI: '{self.iiif_uri}' does not conform to the specifications."
+                )
+                warnings.warn(DspToolsUserWarning(msg))
+            prop_ = etree.Element(
+                f"{DASCH_SCHEMA}iiif-uri",
+                permissions=self.permissions_for_file,
+                nsmap=XML_NAMESPACE_MAP,
+            )
+            prop_.text = self.iiif_uri
+            return prop_
+        return None
+
+    def _serialise_values(self) -> list[etree._Element]:
+        grouped = defaultdict(list)
+        for val in self.values:
+            grouped[val.prop_name].append(val)
+        return [self._combine_values(prop_values) for prop_values in grouped.values()]
+
+    @staticmethod
+    def _combine_values(prop_values: list[Value]) -> etree._Element:
+        prop_ = prop_values[0].make_prop()
+        prop_eles = [x.make_element() for x in prop_values]
+        prop_.extend(prop_eles)
+        return prop_
+
+    def with_IntValue(
```

Comment:
```suggestion
    def with_int_value(
```
naming conventions...